### PR TITLE
refactor file access

### PR DESCRIPTION
**Pull request by andiwand**

### CODE REVIEW

**Comment by andiwand** [comment]
i compared the output with development and it is the same

File: src/OpenDocumentReader.cpp [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    FileMeta meta;
+
+    TranslationConfig config;
+    TranslationContext context;
+
+    OpenDocumentTranslator translatorOd;
+    MicrosoftTranslator translatorMs;
+
+    FileType guess(const std::string &path) noexcept {
+        // TODO guess by file extension first
```
Comment (TomTasche) [inline src/OpenDocumentReader.cpp]: we should never rely on file extensions, on android we don't have them at all in some cases!

Comment (TomTasche) [reply src/OpenDocumentReader.cpp]: i'm using the magic library on android: https://github.com/huzongyao/AndroidMagic

File: src/odf/OpenDocumentMeta.cpp [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+                    .FirstChildElement("meta:document-statistic")
+                    .ToElement();
+            if (statisticsElement != nullptr) {
+                switch (result.type) {
+                    case FileType::OPENDOCUMENT_TEXT: {
+                        const tinyxml2::XMLAttribute *pageCount = statisticsElement->FindAttribute("meta:page-count");
+                        if (pageCount == nullptr) {
+                            break;
+                        }
+                        result.entryCount = pageCount->UnsignedValue();
```
Comment (TomTasche) [inline src/odf/OpenDocumentMeta.cpp]: you're counting pages manually a few lines down. on purpose?

File: src/OpenDocumentReader.cpp (hunk 1 above)
Comment (andiwand) [reply src/OpenDocumentReader.cpp]: but if there is an extension it should be reliable no? i think magic is also using extensions if available, no?

Comment (TomTasche) [reply src/OpenDocumentReader.cpp]: i've seen users trying to open documents with wrong file extensions too... :(

**Pull request merged**